src/main/kotlin/org/dashj/platform/dpp/util/Cbor.kt

```diff
@@ -272,14 +176,14 @@ object Cbor {
             is Boolean -> mapBuilder.put(key, value)
             is Int -> mapBuilder.put(key, value.toLong())
             is BigInteger -> mapBuilder.put(key, value.toLong())
-            is Number -> mapBuilder.put(key, value.toLong())
             is Float -> mapBuilder.put(key, value)
             is Long -> mapBuilder.put(key, value)
             is Double -> mapBuilder.put(key, value)
             is ByteArray -> mapBuilder.put(key, value)
             is Identifier -> mapBuilder.put(key, value.toBuffer())
+            is Number -> mapBuilder.put(key, value.toLong())
             null -> mapBuilder.put(UnicodeString(key), SimpleValue(SimpleValueType.NULL))
-            else -> mapBuilder.put(key, value.toString()) // ?
+            else -> throw IllegalArgumentException("No converter for $value)")
```

Comment:
Having `Number` higher in the list would hide `Long`.  It may not be necessary to have a line for `Number`